File: jvm/core/src/main/java/se/idkollen/client/models/MitIdBackchannelAuthRequest.java
package se.idkollen.client.models;

import org.jspecify.annotations.Nullable;

/**
 * Request body for starting a MitID backchannel authentication session.
 */
public class MitIdBackchannelAuthRequest {
    public String ssn = "";
    public String bindingMessage = "";
    public @Nullable String callbackUrl;
    public @Nullable String refId;

    public MitIdBackchannelAuthRequest setSsn(String v) {
        ssn = v;
        return this;
    }

    public MitIdBackchannelAuthRequest setBindingMessage(String v) {
        bindingMessage = v;
        return this;
    }

    public MitIdBackchannelAuthRequest setCallbackUrl(@Nullable String v) {
        callbackUrl = v;
        return this;
    }

    public MitIdBackchannelAuthRequest setRefId(@Nullable String v) {
        refId = v;
        return this;
    }
}

File: php/src/Models/MitId/MitIdBackchannelAuthRequest.php
<?php

declare(strict_types=1);

namespace Idkollen\Client\Models\MitId;

readonly class MitIdBackchannelAuthRequest
{
    public function __construct(
        public string $ssn,
        public string $bindingMessage,
        public ?string $callbackUrl = null,
        public ?string $refId = null,
    ) {}
}
